manim/mobject/opengl_mobject.py

```diff
@@ -1825,6 +1826,17 @@ def get_shader_uniforms(self):
     def get_shader_vert_indices(self):
         return self.shader_indices
 
+    @property
+    def submobjects(self):
+        return self._submobjects
+
+    @submobjects.setter
+    def submobjects(self, submobject_list):
+        if not hasattr(self, "submobjects"):
+            self._submobjects = submobject_list
```

Comment:
**Question :** I don't understand the meaning of this. How is it possible, since there is a property and `submobjects` is declared in the `__init__` ?

Reply:
> that SGTM, but I think we should get rid of set_submobjects as well.


Yea getting rid of the current setter makes sense, I guess we would have to deprecate it first right?


> Question : I don't understand the meaning of this. How is it possible, since there is a property and submobjects is declared in the __init__

Yes but the submobjects in the init method actually triggers this setter, so in the first case the setter is called it needs to set `self._submobjects` and the cases after that we don't just assign `submobject_list ` to `_submobjects` like most setters would, because it needs to use the setting logic that includes calling the remove method etc. I would be open to a different idea because I wasn't 100% keen on this way of doing it, but wasn't sure of an alternative